packages/rs-drive-abci/src/query/proofs/v0/mod.rs

```diff
@@ -3,501 +3,350 @@ use crate::error::Error;
-                    document_type_name: document_proof_request.document_type,
-                    document_type_keeps_history: document_proof_request.document_type_keeps_history,
-                    document_id: document_id.into_buffer(),
-                    block_time_ms: None, //None because we want latest
-                    contested_status,
-                })
-            })
-            .collect::<Result<Vec<SingleDocumentDriveQuery>, QueryError>>());
-
-        let vote_queries = check_validation_result_with_data!(votes
-            .into_iter()
-            .filter_map(|vote_proof_request| {
-                if let Some(request_type) = vote_proof_request.request_type {
-                    match request_type {
-                        RequestType::ContestedResourceVoteStatusRequest(contested_resource_vote_status_request) => {
-                            let identity_id = match contested_resource_vote_status_request.voter_identifier.try_into() {
-                                Ok(identity_id) => identity_id,
-                                Err(_) => return Some(Err(QueryError::InvalidArgument(
-                                    "voter_identifier must be a valid identifier (32 bytes long)".to_string(),
-                                ))),
-                            };
-                            let contract_id = match contested_resource_vote_status_request.contract_id.try_into() {
-                                Ok(contract_id) => contract_id,
-                                Err(_) => return Some(Err(QueryError::InvalidArgument(
-                                    "contract_id must be a valid identifier (32 bytes long)".to_string(),
-                                ))),
-                            };
-                            let document_type_name = contested_resource_vote_status_request.document_type_name;
-                            let index_name = contested_resource_vote_status_request.index_name;
-                            let index_values = match contested_resource_vote_status_request.index_values.into_iter().enumerate().map(|(pos, serialized_value)|
-                                Ok(bincode::decode_from_slice(serialized_value.as_slice(), bincode::config::standard().with_big_endian()
-                                    .with_no_limit()).map_err(|_| QueryError::InvalidArgument(
-                                    format!("could not convert {:?} to a value in the index values at position {}", serialized_value, pos),
-                                ))?.0)
-                            ).collect::<Result<Vec<_>, QueryError>>() {
-                                Ok(index_values) => index_values,
-                                Err(e) => return Some(Err(e)),
+    ) -> Result<QueryValidationResult<GetProofsResponse>, Error> {
+        let state_transition: StateTransition =
+            StateTransition::deserialize_from_bytes(&state_transition_bytes)?;
```

Comment:
Won't this give an internal error if we can't deserialize it?

Reply:
You right! I will fix it.